### ISSUE TITLE
Previously set Options is overwritten by AddGraphQL()

### ISSUE THREAD

**Issue by serifler** [graphql-dotnet/server#312]
The method **AddGraphQL()** currently overwrites any previously set options, when the expected behaviour would be to add upon previous options.

The following line is responsible for this behaviour:
https://github.com/graphql-dotnet/server/blob/a1d1295d6fefa675f96b119df1f00ada531ff64a/src/Core/ServiceCollectionExtensions.cs#L34

In AspNetCore [RoutingServiceCollectionExtensions.cs](https://github.com/dotnet/aspnetcore/blob/1c126ab773059d6a5899fc29547cb86ed49c46bf/src/Http/Routing/src/DependencyInjection/RoutingServiceCollectionExtensions.cs#L123) it is done with the following line:
`services.Configure(configureOptions);`

The latter allows for configuring options elsewhere in startup like:
```
// first
services.AddOptions()
    .AddOptions<GraphQLOptions>()
    .Bind(configuration.GetSection(nameof(GraphQLOptions))) // from appsettings.json
    .Configure(opt => opt.ComplexityConfiguration.MaxDepth = 1));

// then
services.AddGraphQL(); // currently this ignores all of the above, using the defaults
```

According to the [Options Pattern docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-3.1#configure-simple-options-with-a-delegate):
> Each call to Configure adds an IConfigureOptions<TOptions> service to the service container. In the preceding example, the values of Option1 and Option2 are both specified in appsettings.json, but the values of Option1 and Option2 are overridden by the configured delegate.

> When more than one configuration service is enabled, the last configuration source specified wins and sets the configuration value.

For now I use this workaround:

```
public static IGraphQLBuilder AddGraphQLFixed(
    this IServiceCollection services)
{
    services.TryAddSingleton<IDocumentExecuter, DocumentExecuter>();
    services.TryAddTransient(typeof(IGraphQLExecuter<>), typeof(DefaultGraphQLExecuter<>));
    services.TryAddSingleton<IDocumentWriter>(x =>
    {
        var jsonSerializerSettings = x.GetRequiredService<IOptions<JsonSerializerSettings>>();
        return new DocumentWriter(Formatting.None, jsonSerializerSettings.Value);
    });
    return new GraphQLBuilder(services);
}

public static IGraphQLBuilder AddGraphQLFixed(
    this IServiceCollection services,
    Action<GraphQLOptions> configureOptions)
{
    return services
        .Configure(configureOptions)
        .AddGraphQLFixed();
}
```
Which for example allows me to do:
```
// first
services.AddOptions()
    .AddOptions<GraphQLOptions>()
    .Bind(configuration.GetSection(nameof(GraphQLOptions)))); // from appsettings.json

// then
services.AddGraphQL(opt => opt.ComplexityConfiguration.MaxDepth = 1);
```
I am using the following packages:
```
<TargetFramework>netcoreapp3.1</TargetFramework>
<PackageReference Include="GraphQL" Version="3.0.0-preview-1352" />
<PackageReference Include="GraphQL.Server.Core" Version="3.5.0-alpha0027" />
```

**Comment by sungam3r** [comment]
Will be fixed after merging #299

**Comment by sungam3r** [comment]
@serifler Could you please check if the problem is solved?
